src/dynamic_fees/types.rs

```diff
@@ -0,0 +1,43 @@
+use crate::types::Balance;
+use sp_arithmetic::FixedU128;
+
+/// Fee parameters - minimum and maximum fee, decay and amplification.
+#[derive(Debug, Clone)]
+pub struct FeeParams<Fee> {
+    pub min_fee: Fee,
+    pub max_fee: Fee,
+    pub decay: FixedU128,
+    pub amplification: FixedU128,
+}
+
+/// Oracle entry data for an asset, providing amount in and out and total liquidity of an asset.
+#[derive(Debug, Clone)]
+pub struct OracleEntry {
+    pub amount_in: Balance,
+    pub amount_out: Balance,
+    pub liquidity: Balance,
+}
+
+impl OracleEntry {
+    pub(super) fn net_volume(&self, direction: NetVolumeDirection) -> (Balance, bool) {
```

Comment:
docs, please (specifically describe what the `bool` represents (it's whether the return value is negative, correct?)

any reason to restrict usage to this module?

Reply:
this is internal helper.  no need to expose it.